charts/osm/templates/osm-configmap.yaml

```diff
@@ -13,7 +13,7 @@ data:
 
   tracing_enable: {{ .Values.OpenServiceMesh.tracing.enable | quote }}
 {{- if .Values.OpenServiceMesh.tracing.enable }}
-  tracing_address: {{ .Values.OpenServiceMesh.tracing.address | quote }}
+  tracing_address: {{ include "osm.tracingAddress" . | quote }}
```

Comment:
Doesn't this break BYO Jaeger? e.g. If I already have Jaeger installed behind a service called "my-jaeger" outside the OSM namespace, how would I point to that? I think you could workaround it with the `tpl` function: https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function

Reply:
Can we just make the entire address configurable instead of the namespace?

Reply:
That's how it is today where you just change the `tracing.address` value directly. The problem is the default isn't responsive to when you also set `osmNamespace`, so anytime you override `osmNamespace` you also have to override `tracing.address` if you intend to point to the Jaeger instance installed by the chart.

Reply:
I'm confused as to why this would break BYO Jaeger? The default function makes it so that it only uses the templated value if the value is left empty in `values.yaml` . The assumption is that users will replace the address if they bring up their own instance, which is what is documented under BYO tracing in [tracing.md](https://github.com/openservicemesh/osm/blob/main/docs/content/docs/patterns/observability/tracing.md)

Reply:
After discussing with @eduser25, it seemed like it would make sense to use this template but also set the `values.yaml` value to empty + add a comment in values.yaml and in tracing.md explaining the default. The rationale: this value is only used when `--enable-tracing` is used for the BYO experience but even so, OSM will still not bring up the tracing instance. So there is a default but we expect users to override it

Reply:
Ah, yep. I missed the last `{{ default }}` line in the template. That should work for BYO. Should the default in values.yaml be an empty string then?

Reply:
yes that's what we were thinking of changing it to

Reply:
Yep, sounds good to me. I should've hit refresh before asking that 😄.